storage/rocksdb/ha_rocksdb.cc

```diff
@@ -2334,18 +2334,20 @@ class Rdb_transaction_impl : public Rdb_transaction {
 
   rocksdb::Status get(rocksdb::ColumnFamilyHandle *const column_family,
                       const rocksdb::Slice &key,
-                      std::string *value) const override {
+                      rocksdb::PinnableSlice *const value) const override {
+    value->Reset();
```

Comment:
Once convert_from_storage_format() is called, I think m_retrieved_record needs to be Reset() to free up the underlying memory. I don't think there's a guarantee there will be a follow-up get() call.

For a point query such as "select * from table where pk = k", where does m_retrieved_record->Reset() get called?

Reply:
There is a ha_handler#reset method which calls m_retrieved_record->Reset(). [The documentation](https://github.com/Tema/mysql-5.6/blob/1f2eb46cc766543e39ae1e4e5e05eabb350d8181/sql/handler.h#L3318-L3323) says that it is called after every statement for all tables used by that statement.
So we call Reset() before each Get() to make sure it is cleared for multiple Get() per statement and rely on handler::reset() to clear the latest Get in the statement.

I looked into resetting pinnable slice after convert_from_storage_format() to make it more robust but given that we can't clear it for blobs as a result buffer [holds the pointer to the blob](https://github.com/Tema/mysql-5.6/blob/b8fee8d0de505084d3a34809c72b6952593a370e/storage/rocksdb/ha_rocksdb.cc#L5050-L5052) ([code](https://github.com/Tema/mysql-5.6/blob/b8fee8d0de505084d3a34809c72b6952593a370e/storage/rocksdb/ha_rocksdb.cc#L4978-L4983)), I think relying on handler::reset() is more robust way to go ahead.